crates/ruff_python_formatter/src/expression/expr_lambda.rs

```diff
@@ -95,17 +107,32 @@ impl FormatNodeRule<ExprLambda> for FormatExprLambda {
                 write!(f, [dangling_comments(dangling_before_parameters)])?;
             }
 
-            write!(
-                f,
-                [parameters
-                    .format()
-                    .with_options(ParametersParentheses::Never)]
-            )?;
+            // Try to keep the parameters on a single line, unless there are intervening comments.
+            if preview && !parameters_have_comments {
+                let mut buffer = RemoveSoftLinesBuffer::new(f);
+                write!(
+                    buffer,
+                    [parameters
+                        .format()
+                        .with_options(ParametersParentheses::Never)]
+                )?;
+            } else {
+                write!(
+                    f,
+                    [parameters
+                        .format()
+                        .with_options(ParametersParentheses::Never)]
+                )?;
+            }
 
             write!(f, [token(":")])?;
 
             if dangling_after_parameters.is_empty() {
                 write!(f, [space()])?;
+            }
+            // In preview, always parenthesize the body if there are dangling comments.
+            else if preview {
+                return format_body(body, dangling_after_parameters, self.layout).fmt(f);
```

Comment:
I think `format_body` should be renamed to `parenthesized_body` or `body_with_comments` or what not if this really is about formatting a parenthesized body and not the function to format any lambda body